src/librustc_errors/lib.rs

```diff
@@ -56,6 +56,14 @@ mod lock;
 
 use syntax_pos::{BytePos, Loc, FileLinesResult, FileMap, FileName, MultiSpan, Span, NO_EXPANSION};
 
+#[derive(Copy, Clone, Debug, PartialEq, Hash, RustcEncodable, RustcDecodable)]
+pub enum SuggestionApproximate {
```

Comment:
I feel like this should have a different name given that it seems to include non-approximate (`MachineApplicable`) suggestions. `SuggestionDefiniteness`?

Reply:
I see what you mean. My first suggestion would've been SuggestionApplicability, though. (Let the bikeshedding begin)

Reply:
I like `Applicability`

Reply:
Updated